### PR TITLE
Add events_and_payments.py, with a new list_events() function.

### PR DESCRIPTION
**Pull request by MrPiggy105**
This adds a new file with a new function. Most of the code is similar to the other files, but this required a LOT of filtering of the site_data df columns. Also removed flake8 config as setup.cfg works just as well for me and better for others.

### CODE REVIEW

**Comment by Capta1nT0ad** [comment]
Ideally keep commit messages to 52 chars or less next time. I will review the code.

**Comment by MrPiggy105** [comment]
Ready for review/merge => main. I will deal with attachments later. @Capta1nT0ad 

(not trying to be pushy btw, take your time)

File: src/opentass/events_and_payments.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    df = df.drop("paid_date", errors="ignore")
+    df = df.drop("attach_id", errors="ignore")
+    df = df.drop("surname", errors="ignore")
+    df = df.drop("update_by", errors="ignore")
+    df = df.drop("med_ack_by", errors="ignore")
+    df = df.drop("usi", errors="ignore")
+    df = df.drop("start_date", errors="ignore")
+    df = df.drop("status_code", errors="ignore")
+    df = df.drop("e_mail", errors="ignore")
+    df = df.drop("link_url", errors="ignore")
```
Comment (Capta1nT0ad) [inline src/opentass/events_and_payments.py:68]: Keep this.

File: src/opentass/events_and_payments.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    df = df.drop("id", errors="ignore")
+    df = df.drop("doe", errors="ignore")
+    df = df.drop("dol", errors="ignore")
+    df = df.drop("given_name", errors="ignore")
+    df = df.drop("stud_ack_by", errors="ignore")
+    df = df.drop("preferred_surname", errors="ignore")
+    df = df.drop("other_name", errors="ignore")
+    df = df.drop("entry_code", errors="ignore")
+    df = df.drop("start_time", errors="ignore")
+    df = df.drop("sex", errors="ignore")  # don't even think about it-
```
Comment (Capta1nT0ad) [inline src/opentass/events_and_payments.py:51]: Really!?

File: src/opentass/events_and_payments.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    # OK, there's a lot to filter out here, some of which is basically a duplicate of another column
+    # If you need any use the raw json data or make a pull request if you think people will want it
+
+    df = df.T.drop("paid_flg", errors="ignore")
+    df = df.drop("stud_ack_flg", errors="ignore")
+    df = df.drop("table_id", errors="ignore")
+    df = df.drop("pctut_grp", errors="ignore")
+    df = df.drop("boarder", errors="ignore")
+    df = df.drop("record_id", errors="ignore")
+    df = df.drop("id", errors="ignore")
```
Comment (Capta1nT0ad) [inline src/opentass/events_and_payments.py:42]: Keep this.

**Review by Capta1nT0ad — CHANGES_REQUESTED** [review]
Please address the couple of things I mentioned.

**Comment by MrPiggy105** [comment]
Fixed requested changes.

**Comment by MrPiggy105** [comment]
Fully fixed now!!

**Review by MrPiggy105 — COMMENTED** [review]
Fully done now!

File: src/opentass/events_and_payments.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    )
+
+    site_data = protected_response.json()
+
+    df = pd.DataFrame(site_data["DATA"])
+
+    # OK, there's a lot to filter out here, some of which is basically a duplicate of another column
+    # If you need any use the raw json data or make a pull request if you think people will want it
+
+    df = df.T.drop("paid_flg", errors="ignore")
```
Comment (Capta1nT0ad) [inline src/opentass/events_and_payments.py:36]: Please keep this one, then ready for merge.

**Review by Capta1nT0ad — CHANGES_REQUESTED** [review]
Just one more change and this one is ready for merge.

**Comment by Capta1nT0ad** [comment]
@MrPiggy105 Please review the changes I have requested, then this will be ready to merge.

**Comment by Capta1nT0ad** [comment]
Please fix so I can merge @MrPiggy105.

**Review by MrPiggy105 — COMMENTED** [review]
Done! Ready for merge, can you have a look at my comment on the other PR review? Sorry it took so long for such a short thing

**Comment by Capta1nT0ad** [comment]
@MrPiggy105 It appears that you have not committed your changes.

**Comment by MrPiggy105** [comment]
Whoops... pushed now.

**Comment by Capta1nT0ad** [comment]
Thank you! please now review the other pr.

**Pull request merged**